### ISSUE TITLE
jsdoc/require-jsdoc - publicOnly not working with contexts[].PropertyDefinition

### ISSUE THREAD

**Issue by reececomo** [gajus/eslint-plugin-jsdoc#1122]
tl;dr: Trying to configure a rule so that we document all public members/properties/accessors.

## Expected behavior
When using `jsdoc/require-jsdoc` I would expect `publicOnly` and `contexts[].PropertyDefinition` together to mean that **public properties** required docs.

## Actual behavior
Private/protected members are also failing the lint check too (unexpectedly).

We tried `ClassProperty` too, but that appears to cause a lot of false negatives.

For context there are also these related tickets, but we did not find the answer:
- https://github.com/gajus/eslint-plugin-jsdoc/issues/640
- https://github.com/gajus/eslint-plugin-jsdoc/issues/786
- https://github.com/gajus/eslint-plugin-jsdoc/issues/841

## ESLint Config

```js
    "jsdoc/require-jsdoc": [
      'error',
      {
        checkConstructors: false,
        publicOnly: true,
        contexts: [
          'PropertyDefinition'
        ]
      }
    ],
```

## ESLint sample

```js
export class MyClass {

  public myPublicProperty: number = 1;
       /* ^ Missing JSDoc comment. eslint(jsdoc/require-jsdoc) - expected ✅ */

  private myPrivateProp: number = -1;
       /* ^ Missing JSDoc comment. eslint(jsdoc/require-jsdoc) - unexpected ❌ */

  // ...
}
```

## Environment

- Node version: ^16.16.0
- ESLint version: 8.43.0
- `eslint-plugin-jsdoc` version: 46.4.0

**Comment by reececomo** [comment]
@brettz9 same thing with `require.MethodDefinition` and the other requires 😬

Should I open a separate issue?

**Comment by reececomo** [comment]
Also, thanks for jumping on that so quickly!

**Comment by brettz9** [comment]
I need a code sample regardless of where you post it.

**Comment by reececomo** [comment]
It's the exact same issue but for class methods instead of properties

**Comment by brettz9** [comment]
So your context is exactly `MethodDefinition`?

**Comment by reececomo** [comment]
Moved to https://github.com/gajus/eslint-plugin-jsdoc/issues/1124 😄
